web/src/stream_data.ts

```diff
@@ -591,36 +591,17 @@ export function can_preview(sub: StreamSubscription): boolean {
     return sub.subscribed || !sub.invite_only || sub.previously_subscribed;
 }
```

Comment:
I think a bunch of these little helpers are functionally identical to `has_content_access` and can probably be replaced by it; we can do a sweep in a follow-up, since it'd overcomplicate this PR to do it here.

Reply:
Opened #33400 to track this, marking as resolved for this PR.